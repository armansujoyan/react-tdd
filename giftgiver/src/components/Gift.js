import React, { Component } from 'react';
import { Form, FormGroup, FormControl, ContorlLabel, Button} from 'react-bootstrap';

export default class Gift extends Component {
  constructor(){
    super();

    this.state = {
      person: '',
      present: '',
    }
  }

  render() {
    return (
      <div>
        <Form>
          <FormGroup>
            <ContorlLabel>Person</ContorlLabel>
            <FormControl className='input-person'
              onChange={event => this.setState({ person: event.target.value })}/>
          </FormGroup>
        </Form>
      </div>
    )
  }
}
