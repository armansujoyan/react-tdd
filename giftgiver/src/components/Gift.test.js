import React from 'react';
import { shallow } from 'enzyme';
import Gift from './Gift';

describe('Gift component tests', () => {
  const gift = shallow(<Gift />);

  it('Renders properly', () => {
    expect(gift).toMatchSnapshot();
  });

  it('Initializes a person and present in `state`', () => {
    expect(gift.state()).toEqual({ person: '', present: ''});
  })

  describe('When typing in person input', () => {
    const person = 'Sister';

    beforeEach(() => {
      gift.find('.input-person').simulate('change', { target: { value: person}});
    })

    it('Updates the person in state', () => {
      expect(gift.state().person).toEqual(person);
    })
  })
});