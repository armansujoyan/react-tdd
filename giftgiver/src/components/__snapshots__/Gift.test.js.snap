// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift component tests Renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Form
        bsClass="form"
        componentClass="form"
        horizontal={false}
        inline={false}
      >
        <FormGroup
          bsClass="form-group"
        >
          <UNDEFINED>
            Person
          </UNDEFINED>
          <FormControl
            bsClass="form-control"
            className="input-person"
            componentClass="input"
            onChange={[Function]}
          />
        </FormGroup>
      </Form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "form",
        "children": <FormGroup
          bsClass="form-group"
        >
          <UNDEFINED>
            Person
          </UNDEFINED>
          <FormControl
            bsClass="form-control"
            className="input-person"
            componentClass="input"
            onChange={[Function]}
          />
        </FormGroup>,
        "componentClass": "form",
        "horizontal": false,
        "inline": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": Array [
            <UNDEFINED>
              Person
            </UNDEFINED>,
            <FormControl
              bsClass="form-control"
              className="input-person"
              componentClass="input"
              onChange={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Person",
            },
            "ref": null,
            "rendered": "Person",
            "type": undefined,
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "className": "input-person",
              "componentClass": "input",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
        >
          <FormGroup
            bsClass="form-group"
          >
            <UNDEFINED>
              Person
            </UNDEFINED>
            <FormControl
              bsClass="form-control"
              className="input-person"
              componentClass="input"
              onChange={[Function]}
            />
          </FormGroup>
        </Form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": <FormGroup
            bsClass="form-group"
          >
            <UNDEFINED>
              Person
            </UNDEFINED>
            <FormControl
              bsClass="form-control"
              className="input-person"
              componentClass="input"
              onChange={[Function]}
            />
          </FormGroup>,
          "componentClass": "form",
          "horizontal": false,
          "inline": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <UNDEFINED>
                Person
              </UNDEFINED>,
              <FormControl
                bsClass="form-control"
                className="input-person"
                componentClass="input"
                onChange={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Person",
              },
              "ref": null,
              "rendered": "Person",
              "type": undefined,
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "className": "input-person",
                "componentClass": "input",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
